packages/core/helper-plugin/src/components/SearchURLQuery.tsx

```diff
@@ -1,28 +1,39 @@
-import React, { useLayoutEffect, useRef, useState } from 'react';
+import * as React from 'react';
 
 import { Icon, IconButton, Searchbar, SearchForm } from '@strapi/design-system';
 import { Search as SearchIcon } from '@strapi/icons';
-import PropTypes from 'prop-types';
 import { useIntl } from 'react-intl';
 
-import { useTracking } from '../features/Tracking';
+import { TrackingEvent, useTracking } from '../features/Tracking';
 import { useQueryParams } from '../hooks/useQueryParams';
 
-const SearchURLQuery = ({ label, placeholder, trackedEvent, trackedEventDetails }) => {
-  const inputRef = useRef(null);
-  const iconButtonRef = useRef(null);
+export interface SearchURLQueryProps {
+  label: string;
+  placeholder?: string;
+  trackedEvent?: TrackingEvent['name'] | null;
+  trackedEventDetails?: TrackingEvent['properties'];
+}
+
+const SearchURLQuery = ({
+  label,
+  placeholder = undefined,
+  trackedEvent = null,
```

Comment:
```suggestion
  trackedEvent,
```

IMO we can just use undefined, i'm not sure what benefits of keeping `null` are?